src/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -697,6 +712,74 @@ std::string SQLiteDBEngine::buildLeftOnlyQuery(const std::string& t1,
     return std::string("SELECT "+fieldsList+" FROM "+t1+" t1 LEFT JOIN "+t2+" t2 ON "+onMatchList+" WHERE "+nullFilterList+";");
 } 
 
+void SQLiteDBEngine::getRowDiff(const std::string& table,
+                                const nlohmann::json& data,
+                                nlohmann::json& jsResult)
+{
+    std::vector<std::string> primaryKeyList;
+    if (getPrimaryKeysFromTable(table, primaryKeyList))
+    {
+        const auto& stmt
+        {
+            getStatement(buildSelectMatchingPKsSqlQuery(table, primaryKeyList))
+        };
+
+        const auto& tableFields { m_tableFields[table] };
+        int32_t index { 1l };
+        for (const auto& pkValue : primaryKeyList)
+        {
+            const auto& it
+            {
+                std::find_if(tableFields.begin(), tableFields.end(),
+                                [&pkValue](const ColumnData& column)
+                                {
+                                    return 0 == std::get<Name>(column).compare(pkValue);
+                                })
+            };
+
+            if(it != tableFields.end())
+            {
+                jsResult.push_back({{pkValue, data[pkValue]}});
+                bindJsonData(stmt, *it, data, index);
+                ++index;
+            }
+        }
+
+        if (stmt->step() == SQLITE_ROW)
+        {
+            // The row exist, so lets generate the diff
+            Row regitryFields;
+            for(const auto& field : tableFields)
+            {
+                getTableData(stmt,
+                            std::get<TableHeader::CID>(field),
+                            std::get<TableHeader::Type>(field),
+                            std::get<TableHeader::Name>(field),
+                            regitryFields);
+            }
+
+            if(!regitryFields.empty())
+            {
+                for (const auto& value : regitryFields)
+                {
+                    nlohmann::json object;
+                    if(getFieldValueFromTuple(value, object))
+                    {
+                        if(data[value.first] != object[value.first])
+                        {
+                            jsResult.push_back({{value.first, data[value.first]}});
+                        }
+                    }
+                }
+            }
+        }
+        else
+        {
+            jsResult.clear();
```

Comment:
this should be inside an if statement that checks a isModified flag.

Reply:
Will do.